MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -61,15 +67,24 @@ final class MapViewModel: ObservableObject {
 
     @MainActor private func loadFarmers() async {
         do {
-            let farmers = try await self.farmerService.loadFarmers()
+            progressLoadingOfFarmers = 0.0
```

Comment:
There is too much logic here. How many things are you doing here? Are you still just loading farmers ?